ipvs.go

```diff
@@ -199,23 +199,37 @@ func parseIPVSBackendStatus(file io.Reader) ([]IPVSBackendStatus, error) {
 	return status, nil
 }
 
-func parseIPPort(s string) (net.IP, uint16, error) {
-	tmp := strings.SplitN(s, ":", 2)
-
-	if len(tmp) != 2 {
-		return nil, 0, fmt.Errorf("invalid IP:Port: %s", s)
-	}
+func parseIPPort(ipPort string) (net.IP, uint16, error) {
+	var (
+		portString string
+		ip         net.IP
+		err        error
+	)
 
-	if len(tmp[0]) != 8 && len(tmp[0]) != 32 {
-		return nil, 0, fmt.Errorf("invalid IP: %s", tmp[0])
+	if len(ipPort) == 13 || len(ipPort) == 37 {
+		tmp := strings.SplitN(ipPort, ":", 2)
```

Comment:
Yeah, exactly. I'd find it clearer to only do one thing at a time.

```go
parts := strings.SplitN(s, ":", 2)
if len(parts) != 2 {
  return nil, 0, fmt.Errorf("unexpected IP:Port format: %s", s)
}

port, err := strconv.ParseUint(parts[1], 16, 16)
// ...

switch len(parts[0]) {
// ip handling
}
```

Or first ip and then port handling if you prefer that.